### DISCUSSION TITLE
Do we have any guidance on creating docs for contract?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#285] [Questions] [answered]
### Describe your question in detail.

Let's say I want to add docs for my move contract, is there an example? 
Here's [an entry function](https://github.com/aptos-labs/create-aptos-dapp/blob/7358c5d779439d28efa27f5d4fde8bc08fd2d466/templates/fungible-asset-template/move/sources/launchpad.move#L152-L168) I'm trying to document.  Here's a [view function](https://github.com/aptos-labs/create-aptos-dapp/blob/7358c5d779439d28efa27f5d4fde8bc08fd2d466/templates/fungible-asset-template/move/sources/launchpad.move#L256-L261) I'm trying to document. 

I got a bunch of warnings when running move unit test

```sh
warning: invalid documentation comment
```

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment, marked as answer]
# Documenting view functions
Documentation needs to be attached to the item associated.  This is a little bit more picky than say with something like Rust.

## For view functions

Below is an example of how to use with a view function:

```move
/// My awesome module
module 0x42::example { // double slash can be anywhere

  // Double slash can be anywhere

  /// My awesome constant
  const MY_VALUE: u64 = 5;
   
  /// My awesome error message
  const E_MY_ERROR: u64 = 10;

  #[view]
  /// My awesome view function
  fun show_me_the_money() {
    // ...
  }
}
```

## Comments that will fail

Below here are examples of doc comments `///` that will fail
```move
module 0x42::example {

  /// My awesome view function <- must be below the annotation, right above the thing commented
  #[view]
  fun show_me_the_money() {
    // ...
    /// Within a function
  }

  /// Not attached to anything
}
```

# More details on comments

* `//` Can be used anywhere, denotes a normal comment
* `/* */` Can be used anywhere as a block comment
* `///` Is a doc comment, can only be used directly above a documentable item (e.g. variable, function, module).  All annotations `e.g. #[test]` must be above the comment
* `/** */` Is a doc block comment, similar restrictions to `///`

See: https://aptos.dev/move/book/coding-conventions#comments
